pkg/validations/validate.go

```diff
@@ -106,18 +106,27 @@ func (v NodeValidation) Validate(node yamlmeta.Node, thread *starlark.Thread) []
 			_, isNone := result.(starlark.NoneType)
 			isTrue := bool(result.Truth())
 			// in order to pass, the assertion must return Truthy value or None
-			if !(isNone || isTrue){
+			if !(isNone || isTrue) {
 				failures = append(failures, fmt.Errorf("%s (%s) requires %q (by %s)", key, node.GetPosition().AsCompactString(), r.msg, v.position.AsCompactString()))
 			}
 		}
 	}
 	return failures
 }
 
+// DefaultNullSkipTrue sets the kwarg when_null_skip to true if not set explicitly.
+func (v *NodeValidation) DefaultNullSkipTrue() {
+	if v.kwargs.whenNullSkip == nil {
+		t := true
+		v.kwargs.whenNullSkip = &t
+	}
+}
+
 // shouldValidate uses validationKwargs and the node's value to run checks on the value. If the value satisfies the checks,
 // then the NodeValidation's rules should execute, otherwise the rules will be skipped.
 func (v validationKwargs) shouldValidate(value starlark.Value, thread *starlark.Thread) (bool, error) {
-	if v.whenNullSkip {
```

Comment:
If we hoist the type check for `NoneType` and reuse it, I think it reduces the overall cognitive load:
- I think the first `if` statement can now be combined more easily and stay readable.
  - the parens are not necessary for the compiler, but the humans can more easily parse that the nil-check and value check go together with them in place.
- no need to do the type check twice, so net fewer lines... 


```go
	_, isNone := value.(starlark.NoneType)
	if isNone && (v.whenNullSkip != nil && *v.whenNullSkip) {
		return false, nil
	}

	if v.when != nil {
		result, err := starlark.Call(thread, *v.when, starlark.Tuple{value}, []starlark.Tuple{})
		if err != nil {
			return false, err
		}

		isTrue := bool(result.Truth())
		return isNone || isTrue, nil
	}

```